### PR TITLE
Add support for fetching changed lessons, migrate get_base_url to async

### PR DESCRIPTION
**Pull request by Antoni-Czaplicki**

### CODE REVIEW

**Review by kapi2289 — APPROVED** [review]

File: docs/source/full-docs.rst [hunk 1]
```diff
@@ -87,6 +87,12 @@ Data models
 .. autoclass:: vulcan.data.Lesson
     :members:
 
+.. autoclass:: vulcan.data.ChangedLesson
+    :members:
+
+.. autoclass:: vulcan.data.LessonChange
```
Comment (kuba2k2) [inline docs/source/full-docs.rst:93]: What is a LessonChange?

File: vulcan/_data.py [hunk 2]
```diff
@@ -104,13 +113,36 @@ async def get_lessons(
...
         return Lesson.get(self._api, last_sync, deleted, date_from, date_to, **kwargs)
 
+    async def get_changed_lessons(
+        self,
+        last_sync: datetime = None,
+        deleted=False,
+        date_from=None,
+        date_to=None,
+        **kwargs,
+    ) -> Union[AsyncIterator[Lesson], List[int]]:
```
Comment (kuba2k2) [inline vulcan/_data.py:123]: Getting changed lessons should probably return a ChangedLesson, not a Lesson.

File: vulcan/data/__init__.py [hunk 3]
```diff
@@ -5,5 +5,5 @@
 from ._exam import Exam
 from ._grade import Grade, GradeCategory, GradeColumn
 from ._homework import Homework
-from ._lesson import Lesson, LessonRoom
+from ._lesson import ChangedLesson, Lesson, LessonRoom
```
Comment (kuba2k2) [inline vulcan/data/__init__.py:8]: What about LessonChanges?

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: docs/source/full-docs.rst (hunk 1 above)
Comment (Antoni-Czaplicki) [reply docs/source/full-docs.rst]: This is a typo

File: vulcan/_data.py [hunk 4]
```diff
@@ -104,13 +113,36 @@ async def get_lessons(
...
+        """Yields the student's lessons.
+
+        :param `datetime.datetime` last_sync: date of the last sync,
+            gets only the objects updated since this date
+        :param bool deleted: whether to only get the deleted item IDs
+        :param `datetime.date` date_from: Date, from which to fetch lessons, if not provided
+            it's using the today date (Default value = None)
+        :param `datetime.date` date_to: Date, to which to fetch lessons, if not provided
+            it's using the `date_from` date (Default value = None)
+        :rtype: Union[AsyncIterator[:class:`~vulcan.data.Lesson`], List[int]]
```
Comment (kuba2k2) [inline vulcan/_data.py:133]: The return type applies here as well.

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: vulcan/_data.py (hunk 4 above)
Comment (Antoni-Czaplicki) [reply vulcan/_data.py:133]: Ah, yes - I missed it

**Review by kuba2k2 — APPROVED** [review]

**Pull request merged**